alot/settings/manager.py

```diff
@@ -137,30 +139,59 @@ def read_config(self):
         self._accounts = self._parse_accounts(self._config)
         self._accountmap = self._account_table(self._accounts)
 
-    def _process_xdg_default(self, setting_name):
+    @staticmethod
+    def _expand_environment_and_home(value):
         """
-        Processes setting that uses $XDG_CONFIG_HOME in the default value.
+        Expands environment variables and the home directory (~).
 
-        Setting should be set as None in alot.rc.default.
+        Only acts on string input values.
 
-        Expands home directory (~) and updates the path in _config object.
+        $FOO and ${FOO}-style environment variables are expanded, if they
+        exist. If they do not exist, they are left unchanged.
+        The exception are the following $XDG_* variables that are
+        expanded to fallback values, if they are empty or not set:
+        $XDG_CONFIG_HOME
+        $XDG_CACHE_HOME
 
-        :param setting_name: name of setting to be processed
-        :type setting_name: str
-        :returns: path
-        :rvalue: str
+        :param value: configuration string
+        :type value: str
         """
-        path = self._config.get(setting_name)
-        if path.startswith('$XDG_CONFIG_HOME'):
-            if os.environ.get('XDG_CONFIG_HOME', False):
-                xdg_expanded = os.environ['XDG_CONFIG_HOME']
-            else:
-                xdg_expanded = '~/.config'
-            path = path.replace('$XDG_CONFIG_HOME', xdg_expanded)
+        xdg_vars = {'XDG_CONFIG_HOME': '~/.config',
+                    'XDG_CACHE_HOME': '~/.cache'}
+
+        for xdg_name, fallback in xdg_vars.items():
+            if xdg_name in value:
+                xdg_value = get_xdg_env(xdg_name, fallback)
+                value = value.replace('$%s' % xdg_name, xdg_value)\
```

Comment:
Chaining `replace` calls feels a bit weird, but it is so much more readable and simpler than other variantes I tried (regular expression; some form of a for loop)